parser/src/token.rs

```diff
@@ -752,6 +825,56 @@ mod test {
         );
     }
 
+    #[test]
+    fn hex_literals() {
+        test(
+            r#"0x1f 0xf 0x123 0x001 -0xA 0x"#,
+            vec![
+                (r#"~~~~                   "#, IntLiteral(31)),
+                (r#"    ~~~                "#, IntLiteral(15)),
+                (r#"       ~~~~~           "#, IntLiteral(291)),
+                (r#"            ~~~~~      "#, IntLiteral(1)),
+                (r#"                 ~~~~  "#, IntLiteral(-10)),
+                (r#"                     ~~"#, IntLiteral(0)),
```

Comment:
Should we forbid this?

Reply:
I'd say yes, it seems weird since it is sort of a prefix followed by nothing. Tried it in rust and it complained abot "a number without any digits aswell" so that is some precedence for rejecting it.